src/mpyl/steps/postdeploy/cypress_test.py

```diff
@@ -32,55 +32,43 @@ def execute(self, step_input: Input) -> Output:
-                                      volumes=[
-                                          (volume_path, "/cypress"),
-                                          (os.path.expanduser(cypress_config.kubectl_config_path), "/root/.kube/config")
-                                      ],
-                                      workdir="/cypress")
-        if not isinstance(docker_container, Container):
-            raise TypeError("Docker run command should return a container")
-
+        docker_container = self._get_docker_container(volume_path, cypress_config)
         reports_folder = f"reports/{step_input.project.name}"
         artifact = input_to_artifact(artifact_type=ArtifactType.JUNIT_TESTS, step_input=step_input,
                                      spec={TEST_OUTPUT_PATH_KEY: f"{volume_path}/{reports_folder}",
                                            TEST_RESULTS_URL_KEY: ''})
 
         try:
-            execute_with_stream(logger=self._logger, container=docker_container,
-                                command='bash -c "cp cypress.env.json.example cypress.env.json && '
-                                        f"sed -i 's/acceptance/"
-                                        f"{CypressTest._target_to_test_target(step_input.run_properties.target)}"
-                                        f"/' cypress.env.json && "
-                                        f"sed -i 's/{{PR_NUMBER}}/{step_input.run_properties.versioning.pr_number}/' "
-                                        'cypress.env.json"',
-                                task_name="Preparing env file")
-            execute_with_stream(logger=self._logger, container=docker_container, command="yarn install",
-                                task_name="Running yarn install")
-            execute_with_stream(logger=self._logger, container=docker_container, command="yarn cypress install",
-                                task_name="Installing cypress")
-            execute_with_stream(logger=self._logger, container=docker_container, command="yarn cypress verify",
-                                task_name="Verifying cypress")
-            execute_with_stream(logger=self._logger, container=docker_container, command="yarn tsc",
-                                task_name="Compiling typescript")
-            execute_with_stream(logger=self._logger, container=docker_container, command=f"rm -rf {reports_folder}",
-                                task_name="Remove old report files")
-
-            run_command = f'bash -c "yarn cypress run --spec {specs_string} --reporter-options mochaFile=' \
-                          f'"{reports_folder}/[hash].xml" || true"'
+            self._run_container_preparation_steps(docker_container, step_input, reports_folder)
             record_key = cypress_config.record_key
+            run_command = ''
+
             if record_key:
+                ci_build_id = f"{cypress_config.ci_build_id}-{step_input.project.name}"
+                machines = [1, 2, 3]
```

Comment:
Do we want to make this configurable? Or we always need three processes?

Reply:
There isn't much use to making it configurable, there is an optimal number of threads, which is 3-4 (I actually might change it to 4), and a 'normal' user won't know that. So giving them the opportunity to change it might actually be a bad thing.